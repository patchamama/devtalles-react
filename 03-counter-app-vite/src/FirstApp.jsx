//https://cursos.devtalles.com/courses/take/react-cero-experto/lessons/36057299-proptypes

import PropTypes from 'prop-types'

export const FirstApp = ({ title, subTitle, name }) => {
  //   if (!title) {
  //     throw new Error('El title no existe')
  //   }

  return (
    <>
      <h1 data-testid='test-title'> {title} </h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay subtitulo',
  // title: 'Hola, Soy Goku',
}

export default FirstApp
