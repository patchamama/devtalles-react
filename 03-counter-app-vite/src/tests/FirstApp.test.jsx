import { render } from '@testing-library/react'
import { FirstApp } from '../FirstApp'

describe('Pruebas en el archivo FirstApp.test.jsx', () => {
  test('debe de hacer match con el snapshot', () => {
    const title = 'Hola, Soy Goku'
    const { container } = render(<FirstApp title={title} />)

    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar el titulo en un h1', () => {
    const title = 'Hola, Soy Goku'
    const { container, getByText } = render(<FirstApp title={title} />)

    expect(getByText(title)).toBeTruthy()

    const h1 = container.querySelector('h1')
    expect(h1.innerHTML).toContain(title)
  })
})
