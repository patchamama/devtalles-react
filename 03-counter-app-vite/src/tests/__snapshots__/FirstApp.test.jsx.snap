// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo FirstApp.test.jsx debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     
    Hola, Soy Goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
